tsdb/engine/tsm1/compact_property_test.go

```diff
@@ -0,0 +1,115 @@
+package tsm1_test
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/influxdata/influxdb/tsdb/engine/tsm1"
+)
+
+// CompactionProperty represents a property that compaction groups should satisfy
+type CompactionProperty struct {
+	Name        string
+	Description string
+	Validator   func(allFiles []string, groups []tsm1.CompactionGroup) error
+}
+
+// AdjacentFileProperty validates that compaction groups don't create gaps
+var AdjacentFileProperty = CompactionProperty{
+	Name:        "Adjacency Rule",
+	Description: "Files should not have non-adjacent files within the same compaction level - if files A and C are in the same level group, any file B between them should also be in a group at the same level or higher",
+	Validator:   validateFileAdjacency,
+}
+
+type fileInfo struct {
+	index int
+}
+
+// validateFileAdjacency checks that there are no gaps between compaction groups
+// An adjacency violation occurs when files A and C are in different groups, but file B (between A and C)
+// is also in a different group, creating overlapping or non-contiguous ranges
+func validateFileAdjacency(allFiles []string, groups []tsm1.CompactionGroup) error {
+	var fileMap = make(map[string]fileInfo, len(allFiles))
```

Comment:
Should we sort files?  Tests will surely have the correct order, but if we use this in Production code, does `Glob` have ordering promises?

Reply:
`Glob` actually does not have ordering promises in production. When I was implementing a `psuedo` version of this in the PR where I implemented adjacency checks (https://github.com/influxdata/influxdb/pull/26766/files#diff-e0d6c334f09b26e18b81206ad3b242f2fd1ef09a7b23ea326a13eea3ac916a9aR2478-R2483) I did have to sort, if we were to ever pull this out and utilize it for other than tests I think that it would be beneficial to keep the sorting.

Reply:
> Tests will surely have the correct order

apparently not!